src/java.base/share/classes/jdk/internal/foreign/abi/ppc64/sysv/SysVPPC64leVaList.java

```diff
@@ -155,6 +160,11 @@ private void checkNextArgument(MemoryLayout argLayout, long argByteSize) {
 		}
 	}
 
+	private static long getSmallerStructArgSize(MemorySegment structSegment, MemoryLayout structArgLayout) {
+		return (structSegment.byteSize() > structArgLayout.byteSize()) ?
+							structArgLayout.byteSize() : structSegment.byteSize();
```

Comment:
Am I missing something or could this just be this?
```
    return Math.min(structSegment.byteSize(), structArgLayout.byteSize());
```

Reply:
We expect to get the smaller size suitable for the struct, in such case `Math.min()` should work. I will create a PR for this minor adjustment.

Reply:
PRs are created at https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/521 and https://github.com/ibmruntimes/openj9-openjdk-jdk19/pull/49.